File: search/LinearSearch.js
const linearSearch = (haystack, needle) => {
    for (let i = 0; i < haystack.length; i++) {
        if (haystack[i] === needle)
            return haystack[i];
    }
    return -1;
}


let haystack = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 31, 13, 56];
let needle = 12

console.log(linearSearch(haystack, needle));

File: search/LinearSearch.java
package search;

public class LinearSearch {

    public static int linearSearch(int[] haystack, int needle) {
        for (int i = 0; i < haystack.length; i++) {
            if (haystack[i] == needle)
                return haystack[i];
        }
        return -1;
    }

    public static void main(String[] args) {

        int[] haystack = new int[] { 213, 32, 543, 65, 39202, 43, 567, 245, 245, 90 };
        int needle;
        // ask user for needle
        System.out.print("Enter needle: ");
        needle = Integer.parseInt(System.console().readLine());
        // print result
        System.out.println(linearSearch(haystack, needle));
    }

}
